fairlearn/preprocessing/_information_filter.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+
+
+class InformationFilter(BaseEstimator, TransformerMixin):
+    r"""
+    A component that filters out sensitive correlations in a dataset.
+
+    The `InformationFilter` uses a variant of the Gram-Schmidt process
+    to filter information out of the dataset. This can be useful if you
+    want to filter information out of a dataset because of fairness.
+    To explain how it works: given a training matrix :math:`X` that contains
+    columns :math:`x_1, ..., x_k`. If we assume columns :math:`x_1` and :math:`x_2`
+    to be the sensitive columns then the information-filter will
+    remove information by applying these transformations;
+
+    .. math::
+
+       \\begin{split}
+       v_1 & = x_1 \\\\
+       v_2 & = x_2 - \\frac{x_2 v_1}{v_1 v_1}\\\\
+       v_3 & = x_3 - \\frac{x_k v_1}{v_1 v_1} - \\frac{x_2 v_2}{v_2 v_2}\\\\
+       ... \\\\
+       v_k & = x_k - \\frac{x_k v_1}{v_1 v_1} - \\frac{x_2 v_2}{v_2 v_2}
+       \\end{split}
+
+    Concatenating our vectors (but removing the sensitive ones) gives us
+    a new training matrix :math:`X_{filtered} =  [v_3, ..., v_k]`. The final output
+    is an interpolation between the original dataset (without sensitive columns)
+    and the fitlered dataset.
+
+    .. math::
+
+      X_{\text{tfm}} = \alpha X_{\text{filtered}} + (1-\alpha) X_{\text{orig}}
+
+    :param columns: list of columns to filter out this can be a sequence of either int
+                    (in the case of numpy) or string (in the case of pandas).
+    :param alpha: parameter to control how much to filter, for alpha=1 we filter out
+                  all information while for alpha=0 we don't apply any.
+    :param center: setting to tell if this preprocessing step should center the data for
+                   numerical stability
+    """
+
+    def __init__(self, columns, alpha=1, center=True):
+        self.columns = columns
```

Comment:
A couple of comments here:
* Elsewhere in Fairlearn, we say `sensitive_features` for the sensitive feature data. I wonder if we can use some parallel terminology here. How about `sensitive_feature_ids` ? I don't love the verbosity, but since it's not a forced keyword argument, maybe not a big issue.
* Based on the writeup [linear-information-filter.pdf](https://github.com/fairlearn/fairlearn/files/4761588/linear-information-filter.pdf), I think I convinced myself that the cleanest course of action is to (1) center the sensitive features, but (2) don't center the remaining features. So--I'm now in favor of dropping `center` argument. Thoughts?

Reply:
I do think sklearn deprecated non-keyworded arguments in estimator constructors, something to keep in mind.
That said, I don't mind `sensitive_feature_ids`.

Reply:
Resolved!